### PR TITLE
fix blackbar on saving team settings

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers

### CODE REVIEW

File: shared/actions/teams.js [hunk 1]
```diff
@@ -1366,7 +1371,7 @@ const teamsSaga = function*(): Saga.SagaGenerator<any, any> {
   yield Saga.safeTakeEveryPure(TeamsGen.badgeAppForTeams, _badgeAppForTeams)
   yield Saga.safeTakeEveryPure(RouteTreeGen.switchTo, _onTabChange, null, logError)
   yield Saga.safeTakeEvery(TeamsGen.inviteToTeamByPhone, _inviteToTeamByPhone)
-  yield Saga.safeTakeEveryPure(TeamsGen.setPublicity, _setPublicity, _afterSaveCalls)
+  yield Saga.actionToAction(TeamsGen.setPublicity, _setPublicity)
```
Comment (chrisnojima) [inline shared/actions/teams.js]: another example of why splitting this into 2 functions was problematic

File: shared/actions/teams.js [hunk 2]
```diff
@@ -958,102 +947,118 @@ const _setMemberPublicity = function*(action: TeamsGen.SetMemberPublicityPayload
...
-  const openTeamRole = Constants.teamRoleByEnum[teamSettings.joinAs]
-  const publicityAnyMember = teamPublicitySettings.anyMemberShowcase
-  const publicityMember = teamPublicitySettings.member
-  const publicityTeam = teamPublicitySettings.team
-
-  const calls = []
-  if (openTeam !== settings.openTeam || (settings.openTeam && openTeamRole !== settings.openTeamRole)) {
-    calls.push(
-      Saga.call(function*() {
-        RPCTypes.teamsTeamSetSettingsRpcPromise({
```
Comment (chrisnojima) [inline shared/actions/teams.js]: this wasn't returning so it was always undefined

File: shared/actions/teams.js [hunk 3]
```diff
@@ -958,102 +947,118 @@ const _setMemberPublicity = function*(action: TeamsGen.SetMemberPublicityPayload
...
+          )
+            .then(payload => ({payload, type: 'ok'}))
+            .catch(payload => ({payload, type: 'error'}))
         })
-          .then(payload => ({type: 'ok', payload}))
-          .catch(payload => ({type: 'error', payload}))
-      })
-    )
-  }
-  return Saga.all([
```
Comment (chrisnojima) [inline shared/actions/teams.js]: doesn't actually make sense that this was .all. this means the increment and calls and the last call were all racing for some reason

File: shared/actions/teams.js [hunk 4]
```diff
@@ -958,102 +947,118 @@ const _setMemberPublicity = function*(action: TeamsGen.SetMemberPublicityPayload
...
+    }
+    if (publicityTeam !== settings.publicityTeam) {
+      calls.push(
+        Saga.call(function*() {
+          return RPCTypes.teamsSetTeamShowcaseRpcPromise(
+            {
+              isShowcased: settings.publicityTeam,
+              name: teamname,
+            },
+            waitingKey
```
Comment (chrisnojima) [inline shared/actions/teams.js]: pass the waitingkey to all the individual rpcs

File: shared/actions/teams.js [hunk 5]
```diff
@@ -958,102 +947,118 @@ const _setMemberPublicity = function*(action: TeamsGen.SetMemberPublicityPayload
...
+      )
+    }
+
+    const results = yield Saga.all(calls)
+    // TODO delete this getDetails call when CORE-7125 is finished
+    Saga.put(TeamsGen.createGetDetails({teamname}))
+
+    // Display any errors from the rpcs
+    const errs = results
+      .filter(r => r.type === 'error')
```
Comment (chrisnojima) [inline shared/actions/teams.js]: should be error, not err

File: shared/actions/teams.js [hunk 6]
```diff
@@ -958,102 +947,118 @@ const _setMemberPublicity = function*(action: TeamsGen.SetMemberPublicityPayload
...
+      .map(({payload}) => Saga.put(ConfigGen.createGlobalError({globalError: convertToError(payload)})))
+    yield Saga.all(errs)
```
Comment (akalin-keybase) [inline shared/actions/teams.js]: this seems weird...we only have one globalError so if we have more than one error from the calls, we race them? I guess it was what the old code was doing, though

Comment (chrisnojima) [reply shared/actions/teams.js]: yeah, but they all get logged so we shouldn't drop them

**Pull request merged**